### PR TITLE
Feature/assert location

### PR DESCRIPTION
**Pull request by bryzettler**
#345 

Adds the ability to update/assert location of a hotspot from the collectibles screen. Feature fully utilizes Mapbox for map and forward/reverse geolocation search. 

Map centers on 
```javascript 
userLocation || iotLocation || mobileLocation || San Francisco
```

And then creates a bounding box around all those and goes to the center. Not sure if this is the best UX.

https://github.com/helium/wallet-app/assets/694011/14a8c09f-4b7c-4299-9983-3e29a8cec35a

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1 +1,18 @@
-GRAPH_URI=http://localhost:4000/graphiql
\ No newline at end of file
+GRAPH_URI=
+ONE_SIGNAL_APP_ID=
+ONE_SIGNAL_ACCOUNT_TAG_SALT=
+WALLET_CONNECT_PROJECT_ID=
+WALLET_CONNECT_METADATA_URL=
+MAINNET_RPC_URL=
+DEVNET_RPC_URL=
+SOLANA_PREVIEW=
```
Comment (abhay) [inline .env.sample:8]: is this one still needed?

File: src/components/SearchInput.tsx [hunk 2]
```diff
@@ -26,21 +30,27 @@ const SearchInput = ({
...
       paddingStart={padding as Spacing}
       flexDirection="row"
       alignItems="center"
+      {...boxProps}
     >
       <Search color={colors[color as Color]} />
       <TextInput
         textInputProps={{
           onChangeText,
+          onSubmitEditing: () => {
```
Comment (Perronef5) [inline src/components/SearchInput.tsx:44]: Could put this in a `useCallback`

File: src/features/collectables/AssertLocationScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,656 @@
...
+                    borderRadius="round"
+                  />
+                </Box>
+                <Box flexDirection="row" alignItems="center" marginRight="m">
+                  <Text
+                    variant="subtitle4"
+                    color="secondaryText"
+                    marginRight="s"
+                  >
+                    MOBILE:
```
Comment (Perronef5) [inline src/features/collectables/AssertLocationScreen.tsx:361]: Add to translations file?

File: src/features/collectables/AssertLocationScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,656 @@
...
+          {
+            text: t('generic.cancel'),
+            style: 'destructive',
+          },
+        ],
+      )
+    } else {
+      // elevGainVisible
+      // we can assume user is asserting location from elevGain UI
+      await assertLocation('iot')
```
Comment (Perronef5) [inline src/features/collectables/AssertLocationScreen.tsx:290]: Should we navigate back to the hotspot details screen when successful?

**Review by Perronef5 — APPROVED** [review]
Overall LGTM. Just some minor comments

File: .env.sample (hunk 1 above)
Comment (bryzettler) [reply .env.sample:8]: this whole file?

File: src/hooks/useEntityKey.ts
[the remark this answers is not in the document]
Comment (bryzettler) [reply src/hooks/useEntityKey.ts:1]: there already is a `useHotspot` hook. Move this logic into that?

File: src/features/collectables/AssertLocationScreen.tsx (hunk 4 above)
Comment (bryzettler) [reply src/features/collectables/AssertLocationScreen.tsx:290]: Solid UX question. I think the visual cue of the dot being updated on the map is nice feedback. Might lose that if we navigate back to the details screen? But maybe if we display the location on the details screen that would be good feedback too.

File: .env.sample (hunk 1 above)
Comment (ChewingGlass) [reply .env.sample:8]: He's talking about the SOLANA_PREVIEW var

File: src/utils/solanaUtils.ts [hunk 5]
```diff
@@ -1026,37 +1043,51 @@ export const getCompressedNFTMetadata = async (
...
-    const hotspotWithMeta: HotspotWithPendingRewards =
-      hotspot as HotspotWithPendingRewards
-    hotspotWithMeta.pendingRewards = {
-      [Mints.MOBILE]: mobileRewards[entityKeys[index]],
-      [Mints.IOT]: iotRewards[entityKeys[index]],
-    }
-    return hotspotWithMeta
-  })
+  return Promise.all(
+    hotspots.map(async (hotspot, index) => {
```
Comment (ChewingGlass) [inline src/utils/solanaUtils.ts:1079]: Why does this need to be async?

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

**Pull request merged**